src/libraries/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.cs

```diff
@@ -42,18 +42,19 @@ public sealed partial class SafeSocketHandle : SafeHandleMinusOneIsInvalid
         /// <param name="preexistingHandle">Handle to wrap</param>
         /// <param name="ownsHandle">Whether to control the handle lifetime</param>
         public SafeSocketHandle(IntPtr preexistingHandle, bool ownsHandle)
-            : base(ownsHandle)
+            : base(ownsHandle: true) // To support canceling on-going operations we need to detect
```

Comment:
I think it would be wise to add tests verifying that a non-owned handle will not be closed by Dispose and finalization. (Couldn't find existing ones, maybe just missed them?)

Reply:
That makes sense. I'll add those.